web_client/panels/ZoomWidget.js

```diff
@@ -166,6 +178,77 @@ var ZoomWidget = Panel.extend({
         this._zoomSliderChange();
     },
 
+     /**
+     * A handler called when the download view button is clicked.
+     */
+    _downloadView(evt) {
+        var imageId = router.getQuery('image');
+        var bounds = router.getQuery('bounds');
+        var boundsTab = bounds.split(',');
+        var bound;
+        for (bound in boundsTab) {
+            if (boundsTab[bound] < 0) {
+                boundsTab[bound] = 0;
+            }
+        }
+        var params = $.param({
+            width: window.innerWidth,
+            height: window.innerHeight,
+            left: boundsTab[0],
+            top: boundsTab[1],
+            right: boundsTab[2],
+            bottom: boundsTab[3],
+            contentDisposition: 'attachment'
+        });
+        var urlView = `/${apiRoot}/item/${imageId}/tiles/region?${params}`;
+        if (this._cancelSelection === true) {
+            this.viewer.annotationLayer.mode(null);
+            this._cancelSelection = false;
+            $('.h-download-button-area').css('background-color', '');
+        }
+        var hrefAttrView = this.$('a.h-download-link#download-view-link').attr('href');
+        if (typeof hrefAttrView === typeof undefined || hrefAttrView !== urlView) {
+            this.$('a.h-download-link#download-view-link').attr({
+                href: urlView
+            });
+        }
```

Comment:
Setting an attribute is cheap.  I would just set it without trying to detect if it needs to be updated.